### ISSUE TITLE
An Issue of Estimated Rows

### ISSUE THREAD

**Issue by bajinsheng** [cockroachdb/cockroach#89161]
**Describe the problem**

Equivalent `WHERE` predicate return different estimated rows.

**To Reproduce**

```
CREATE TABLE t0 (c0 INT);
INSERT INTO t0 VALUES (1), (2), (3), (4), (5), (6), (7), (8), (9), (10);
ANALYZE t0;

EXPLAIN (VERBOSE) SELECT t0.c0 FROM t0 WHERE (t0.c0 IS NOT NULL); -- estimated rows: 10
EXPLAIN (VERBOSE) SELECT t0.c0 FROM t0 WHERE (t0.c0 IS NOT NULL) OR (1 < ALL (t0.c0, t0.c0)); -- estimated rows: 3
```

**Expected behavior**
I think the estimated rows should be: `(t0.c0 IS NOT NULL)` <= `(t0.c0 IS NOT NULL) OR (1 < ALL (t0.c0, t0.c0))` ?
In this case, both predicates are equivalent, so the estimated rows should also be equal?

**Environment:**
 - CockroachDB version [f188d21d1853ca8b81d9c4f0ab46d7c0fa34ee44]
 - Server OS: [Ubuntu 20.04]
 - Client app [`cockroach sql`]

**Additional context**

Jira issue: CRDB-20152

**Comment by msirek** [comment]
This is a duplicate of #75090.  This may be easy to implement, so it may be good to get this in quickly.

**Comment by msirek** [comment]
Hi @bajinsheng, thanks for your report. This is a known issue #75090 which will be fixed by #89358.
Closing this issue as a dup of #75090.

**Comment by bajinsheng** [comment]
Thanks for your reply! We are exploring some methods to find potential issues in cardinality estimators. Hope we can provide some help!
